ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+            $data['fields']['customfield_' . $jira_field_id] = $element['#options'][$form_state['values'][$key]];
+          }
+          else {
+            $data['fields']['customfield_' . $jira_field_id] = $form_state['values'][$key];
+          }
+        }
+      }
+    }
+  }
+
+  $requestParts = _ereolen_support_get_request_parts('POST', $data);
+
+  $result = drupal_http_request($requestParts['jira_path'] . '/issue', $requestParts['options']);
+  _ereolen_support_log_error($result);
+}
+
+/**
+ * Create the parts required to make a drupal_http_request.
+ *
+ * @param string $method_type
+ *   The request method.
+ * @param array $data
+ *   An optional data array.
+ *
+ * @return array
+ *   An array containing base path for the request and the options related to
+ *   the request.
+ *
+ * @return array
+ */
+function _ereolen_support_get_request_parts($method_type, $data = []) {
+  $jiraUsername = variable_get('jira_username');
+  $jiraPassword = variable_get('jira_password');
+  // Jira requires base64_encode auth.
+  $encoded_auth = base64_encode($jiraUsername . ':' . $jiraPassword);
+  $jira_api_path = variable_get('jira_domain') . '/rest/api/latest';
+
+  $options = array(
+    'method' => $method_type,
+    'timeout' => 15,
+    'headers' => array(
+      'Content-Type' => 'application/json',
+      'Authorization' => 'Basic '. $encoded_auth
+    ),
+  );
+
+  // Add data variable to POST and PUT requests.
+  if ($method_type == 'POST' || $method_type == 'PUT' ) {
+    $options['data'] = json_encode($data);
+  }
+
+  $requestParts = array(
+    'jira_path' => $jira_api_path,
+    'options' => $options
+  );
+
+  return $requestParts;
+}
+
+/**
+ * Prepare an array for use in a select list.
+ *
+ * @param string $vocab_machine_name
+ *  A vocabulary machine name.
+ * @return array
+ *   A list of term names.
+ */
+function _ereolen_support_create_select_list($vocab_machine_name) {
+  $vacabulary = taxonomy_vocabulary_machine_name_load($vocab_machine_name);
+  $terms = taxonomy_get_tree($vacabulary->vid);
+  if ($terms) {
+    $list = [];
+    foreach ($terms as $term) {
+      $list[$term->tid] = str_repeat('-', $term->depth) . $term->name;
+    }
+
+    return $list;
+  }
+  return [];
+}
+
+/**
+ * Provide a support ID grabbed from Drupal user.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ * @return null
+ */
+function _ereolen_support_get_support_id($user) {
+  // @todo If user is logged in provide a Support id.
+  return NULL;
+}
+
+/**
+ * Add an entry to the eReolen support error log.
+ *
+ * @param \stdClass $result
+ *   The result of a jira http request.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_log_error($result) {
+  if ($result->code !== '200' && $result->code !== '201') {
+    $id = db_insert('ereolen_support_errors')
+      ->fields(array(
+        'timestamp' => format_date(time()),
+        'request' => $result->request,
+        'data' => serialize($result->data),
```

Comment:
Are we always sure that the data don't contains disclosed information?

Reply:
The data is whatever the user submitted so... no. But it's no different from what would be stored in Jira had the request not resulted in an error.

Reply:
There are a BIG different... if uses submit CPR and stuff here it should never be store in the Drupal database if not sanitised. As we are not allowed to store that information and if we really have to there have to be documented process for GDPR request and process for deletion and data break in plans etc. Also the site is not hosted in AAK's secure network. And we are not allowed to store even required data more that 30 days, so this would also require automatically deletions etc. 

The site has plans to switch to adgangplatformen to eliminate all forms for user identifications in the system.

Reply:
- Added @todo for form validation on submit to check for CPR.
- Added todo for hook_cron to clear ereolen_support_errors after 30 days.

I'm unsure if that will do it or do we have to document something as you mention?

Reply:
We can ask the PL and production owner as it's those how have to write process in relation to audits and GDPR rules

Reply:
Added CPR check and hook_cron. PL is on top of GDPR rules